util/src/main/java/tech/pegasys/teku/util/config/PortAvailability.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.util.config;
+
+import java.io.IOException;
+import java.net.DatagramSocket;
+import java.net.ServerSocket;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PortAvailability {
+  private static final Logger LOG = LogManager.getLogger();
+
+  public static boolean isPortAvailableForTcp(final int port) {
+    if (!isPortValid(port)) {
+      return false;
+    }
+    try (final ServerSocket serverSocket = new ServerSocket(port)) {
+      serverSocket.setReuseAddress(true);
+      return true;
+    } catch (IOException ex) {
+      LOG.trace("failed to open port for TCP", ex);
+    }
+    return false;
+  }
+
+  public static boolean isPortAvailableForUdp(final int port) {
+    if (!isPortValid(port)) {
+      return false;
+    }
+    try (final DatagramSocket datagramSocket = new DatagramSocket(port)) {
+      datagramSocket.setReuseAddress(true);
```

Comment:
Looking at the docs for `setReuseAddress` it looks like it needs to be set before the socket is bound to guarantee it works.  So we'd have to do something like:
```suggestion
    try (final DatagramSocket datagramSocket = new DatagramSocket()) {
      datagramSocket.setReuseAddress(true);
      datagramSocket.bind(new InetSocketAddress(port));
```

and the same above for TCP.